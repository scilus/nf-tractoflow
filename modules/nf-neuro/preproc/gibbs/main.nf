process PREPROC_GIBBS {
    tag "$meta.id"
    label 'process_single'

    container "mrtrix3/mrtrix3:latest"

    input:
    tuple val(meta), path(dwi)

    output:
    tuple val(meta), path("*dwi_gibbs_corrected.nii.gz"), emit: dwi
    path "versions.yml"                                 , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    export ITK_GLOBAL_DEFAULT_NUMBER_OF_THREADS=1
    export OMP_NUM_THREADS=1
    export OPENBLAS_NUM_THREADS=1
    mrdegibbs $dwi ${prefix}__dwi_gibbs_corrected.nii.gz -nthreads 1

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        mrtrix: \$(mrdegibbs -version 2>&1 | sed -n 's/== mrdegibbs \\([0-9.]\\+\\).*/\\1/p')
    END_VERSIONS
    """

    stub:
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    mrdegibbs -h

    touch ${prefix}__dwi_gibbs_corrected.nii.gz

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        mrtrix: \$(mrdegibbs -version 2>&1 | sed -n 's/== mrdegibbs \\([0-9.]\\+\\).*/\\1/p')
    END_VERSIONS
    """
}
