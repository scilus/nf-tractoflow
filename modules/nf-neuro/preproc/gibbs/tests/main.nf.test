nextflow_process {

    name "Test Process PREPROC_GIBBS"
    script "../main.nf"
    process "PREPROC_GIBBS"

    tag "modules"
    tag "modules_nfcore"
    tag "preproc"
    tag "preproc/gibbs"

    tag "subworkflows"
    tag "subworkflows/load_test_data"

    config "./nextflow.config"

    setup {
            run("LOAD_TEST_DATA", alias: "LOAD_DATA") {
                script "../../../../../subworkflows/nf-neuro/load_test_data/main.nf"
                process {
                    """
                    input[0] = Channel.from( [ "topup_eddy_light.zip" ] )
                    input[1] = "test.load-test-data"
                    """
                }
            }
        }

    test("preproc - gibbs") {
        when {
            process {
                """
                input[0] = LOAD_DATA.out.test_data_directory
                    .map{ test_data_directory -> [
                    [ id:'test', single_end:false ], // meta map
                    file("\${test_data_directory}/sub-01_dir-AP_dwi.nii.gz", checkIfExists: true)
                    ]}
                """
            }
        }
        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }
}
