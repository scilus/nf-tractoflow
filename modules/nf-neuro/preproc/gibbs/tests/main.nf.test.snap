{
    "preproc - gibbs": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__dwi_gibbs_corrected.nii.gz:md5,f15ce978797cd4342089e16204a40436"
                    ]
                ],
                "1": [
                    "versions.yml:md5,65c58dca4630ca36c724dd823f437585"
                ],
                "dwi": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__dwi_gibbs_corrected.nii.gz:md5,f15ce978797cd4342089e16204a40436"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,65c58dca4630ca36c724dd823f437585"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-05-08T17:18:35.063226"
    }
}